streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -566,10 +567,22 @@ void runLoop() {
                 }
             } catch (final TaskMigratedException e) {
                 handleTaskMigrated(e);
+            } catch (final ShutdownRequestedException e){
+                handleShutdownRequest(e);
```

Comment:
The interruption in this location caused ...


```
TaskManager
	MetadataState:
	Tasks:
 died (org.apache.zookeeper.server.NIOServerCnxnFactory:92)
java.lang.IllegalStateException: Illegal state SUSPENDED while completing restoration for active task 0_0
	at org.apache.kafka.streams.processor.internals.StreamTask.completeRestoration(StreamTask.java:245)
	at org.apache.kafka.streams.processor.internals.TaskManager.tryToCompleteRestoration(TaskManager.java:487)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:675)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:554)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:513)
```

Not sure that is okay, the test still runs. If its is shutting down maybe it does matter and I'm not sure where else to put it

Reply:
Interesting. This looks like it's probably just a consequence of the way the test is set up. Some debugging should clear it up.